checks.d/docker_daemon.py

```diff
@@ -0,0 +1,640 @@
+# stdlib
+import os
+import re
+import requests
+import time
+import socket
+import urllib2
+from collections import defaultdict, Counter, deque
+
+# project
+from checks import AgentCheck
+from config import _is_affirmative
+from utils.dockerutil import find_cgroup, find_cgroup_filename_pattern, get_client, MountException, set_docker_settings
+from utils.platform import Platform
+
+
+EVENT_TYPE = 'docker'
+SERVICE_CHECK_NAME = 'docker.service_up'
+SIZE_REFRESH_RATE = 5 # Collect container sizes every 5 iterations of the check
+
+GAUGE = AgentCheck.gauge
+RATE = AgentCheck.rate
+
+CGROUP_METRICS = [
+    {
+        "cgroup": "memory",
+        "file": "memory.stat",
+        "metrics": {
+            "cache": ("docker.mem.cache", GAUGE),
+            "rss": ("docker.mem.rss", GAUGE),
+            "swap": ("docker.mem.swap", GAUGE),
+        },
+        "to_compute": {
+            # We only get these metrics if they are properly set, i.e. they are a "reasonable" value
+            "docker.mem.limit": (["hierarchical_memory_limit"], lambda x: float(x) if float(x) < 2 ** 60 else None, GAUGE),
+            "docker.mem.sw_limit": (["hierarchical_memsw_limit"], lambda x: float(x) if float(x) < 2 ** 60 else None, GAUGE),
+            "docker.mem.in_use": (["rss", "hierarchical_memory_limit"], lambda x,y: float(x)/float(y) if float(y) < 2 ** 60 else None, GAUGE),
+            "docker.mem.sw_in_use": (["swap", "rss", "hierarchical_memsw_limit"], lambda x,y,z: float(x + y)/float(z) if float(z) < 2 ** 60 else None, GAUGE)
+
+        }
+    },
+    {
+        "cgroup": "cpuacct",
+        "file": "cpuacct.stat",
+        "metrics": {
+            "user": ("docker.cpu.user", RATE),
+            "system": ("docker.cpu.system", RATE),
+        },
+    },
+    {
+        "cgroup": "blkio",
+        "file": 'blkio.throttle.io_service_bytes',
+        "metrics": {
+            "io_read": ("docker.io.read_bytes", RATE),
+            "io_write": ("docker.io.write_bytes", RATE),
+        },
+    },
+]
+
+DEFAULT_CONTAINER_TAGS = [
+    "docker_image",
```

Comment:
I think it makes sense to keep it so we don't break everything (dashboards monitors) and as you said it's the same cardinality as image_name / image_tag so it's not creating more contexts, i think it's a good compromise.